packages/api/lambdas/sf-event-sqs-to-db-records.js

```diff
@@ -82,50 +119,124 @@ const shouldWriteExecutionToRDS = async (cumulusMessage, knex) => {
     return executionModel.storeExecutionFromCumulusMessage(cumulusMessage);
-  }
+  });
 
-  try {
-    return await knex.transaction(async (trx) => {
-      await trx(tableNames.executions)
-        .insert({
-          arn: executionArn,
-          cumulus_version: getMessageCumulusVersion(cumulusMessage),
-        });
-      return executionModel.storeExecutionFromCumulusMessage(cumulusMessage);
+const savePdrViaTransaction = async ({
+  cumulusMessage,
+  collection,
+  provider,
+  trx,
+}) =>
+  trx(tableNames.pdrs)
+    .insert({
+      name: getMessagePdrName(cumulusMessage),
+      status: getWorkflowStatus(cumulusMessage),
+      collectionCumulusId: collection.cumulusId,
+      providerCumulusId: provider.cumulusId,
     });
-  } catch (error) {
-    log.error(`Failed to write execution records for ${executionArn}`, error);
-    throw error;
+
+const savePdr = async ({
+  cumulusMessage,
+  collection,
+  provider,
+  knex,
+  pdrModel = new Pdr(),
+}) => {
+  // If there is no PDR in the message, then there's nothing to do here, which is fine
+  if (!messageHasPdr(cumulusMessage)) {
+    return true;
+  }
+  if (!isRecordDefined(collection)) {
+    throw new Error(`Collection reference is required for a PDR, got ${collection}`);
   }
+  if (!isRecordDefined(provider)) {
+    throw new Error(`Provider reference is required for a PDR, got ${provider}`);
+  }
+  return knex.transaction(async (trx) => {
+    await savePdrViaTransaction({ cumulusMessage, collection, provider, trx });
+    return pdrModel.storePdrFromCumulusMessage(cumulusMessage);
+  });
 };
 
-const savePdrToDb = async (cumulusMessage) => {
-  const pdrModel = new Pdr();
+const saveGranulesToDb = async (cumulusMessage) => {
+  const granuleModel = new Granule();
+
   try {
-    await pdrModel.storePdrFromCumulusMessage(cumulusMessage);
+    await granuleModel.storeGranulesFromCumulusMessage(cumulusMessage);
   } catch (error) {
     const executionArn = getMessageExecutionArn(cumulusMessage);
-    log.fatal(`Failed to create/update PDR database record for execution ${executionArn}: ${error.message}`);
+    log.fatal(`Failed to create/update granule records for execution ${executionArn}: ${error.message}`);
     throw error;
   }
 };
 
-const saveGranulesToDb = async (cumulusMessage) => {
+const saveRecordsToDynamoDb = async (cumulusMessage) => {
+  const executionModel = new Execution();
+  const pdrModel = new Pdr();
   const granuleModel = new Granule();
 
+  const results = await Promise.allSettled([
+    executionModel.storeExecutionFromCumulusMessage(cumulusMessage),
+    pdrModel.storePdrFromCumulusMessage(cumulusMessage),
+    granuleModel.storeGranulesFromCumulusMessage(cumulusMessage),
+  ]);
+  const failures = results.filter((result) => result.status === 'rejected');
+  if (failures.length > 0) {
+    const allFailures = failures.map((failure) => failure.reason);
+    log.error(allFailures.join(' '));
```

Comment:
I am not proud of this error handling. I just wanted a way to cleanly log any errors that were encountered. Thoughts?

Reply:
Are there any other statuses besides `rejected` that indicate a failure?

Reply:
Not as far as I know: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled

Reply:
That's a reasonable change to this, presuming that all of the models reject on failures, which they appear to.    I'm not aware of anything cleaner short of a function abstraction that would accomplish this.  

It's worth noting (without suggesting action)  though, in the case of a granule failure, I believe you're only going to get the *first* failed granule from storeGranulesFromCumulusMessage as the model is returning 

```
    return Promise.all(granuleRecords.map(this._validateAndStoreGranuleRecord, this));
```

Reply:
Yeah, we're going to have to refactor the granule write logic which I noticed. See the discussion in the cumulus-ix channel about a transaction for writing all granules vs a transaction per granule. But I'm going to leave that for the next PR

Reply:
I realize we were comfortable with the error handling, but I didn't like the awkward merging of error messages, so I updated this code to throw an AggregateError instead:

093da4197c7569124a851f2eba866505498c51b7